code/controllers/subsystems/dbcore.dm

```diff
@@ -0,0 +1,554 @@
+SUBSYSTEM_DEF(dbcore)
+	name = "Database"
+	flags = SS_TICKER
+	wait = 10 // Not seconds because we're running on SS_TICKER
+	runlevels = RUNLEVEL_INIT|RUNLEVEL_LOBBY|RUNLEVELS_DEFAULT
+	init_order = SS_INIT_DBCORE
+	priority = SS_PRIORITY_DBCORE
+
+	var/failed_connection_timeout = 0
+
+	var/schema_mismatch = 0
+	var/db_minor = 0
+	var/db_major = 0
+	var/failed_connections = 0
+
+	var/last_error
+
+	var/max_concurrent_queries = 25
+
+	///! NOTE: SStime_track is not in yet, but these are essential to other functions.
+	/// Number of all queries, reset to 0 when logged in SStime_track. Used by SStime_track
+	var/all_queries_num = 0
+	/// Number of active queries, reset to 0 when logged in SStime_track. Used by SStime_track
+	var/queries_active_num = 0
+	/// Number of standby queries, reset to 0 when logged in SStime_track. Used by SStime_track
+	var/queries_standby_num = 0
+
+
+	/// All the current queries that exist.
+	var/list/all_queries = list()
+	/// Queries being checked for timeouts.
+	var/list/processing_queries
+
+	/// Queries currently being handled by database driver
+	var/list/datum/db_query/queries_active = list()
+	/// Queries pending execution that will be handled this controller firing
+	var/list/datum/db_query/queries_new
+	/// Queries pending execution, mapped to complete arguments
+	var/list/datum/db_query/queries_standby = list()
+	/// Queries left to handle during controller firing
+	var/list/datum/db_query/queries_current
+
+	var/connection  // Arbitrary handle returned from rust_g.
+
+/datum/controller/subsystem/dbcore/Initialize()
+	//We send warnings to the admins during subsystem init, as the clients will be New'd and messages
+	//will queue properly with goonchat
+	switch(schema_mismatch)
+		if(1)
+			message_admins("Database schema ([db_major].[db_minor]) doesn't match the latest schema version ([DB_MAJOR_VERSION].[DB_MINOR_VERSION]), this may lead to undefined behaviour or errors")
```

Comment:
```suggestion
			message_admins("Database schema ([db_major].[db_minor]) doesn't match the latest schema version ([DB_MAJOR_VERSION].[DB_MINOR_VERSION]), this may lead to undefined behaviour or errors.")
```

Reply:
👍